activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -856,6 +851,12 @@ def connect
           @raw_connection = self.class.new_client(@connection_parameters)
         end
 
+        def reconnect
+          @raw_connection.reset
```

Comment:
`@raw_connection` can't possibly be `nil` here?

Reply:
Not yet (or at least no more than it was in `reconnect!` before); that becomes a factor in #44591, which I split out to make my life more complicated 😅